src/main.ts

```diff
@@ -721,7 +722,7 @@ function generateDecode(ctx: Context, fullName: string, messageDesc: DescriptorP
         }
       } else if (isEnum(field)) {
         if (options.stringEnums) {
-          const fromJson = getEnumMethod(typeMap, field.typeName, 'FromJSON');
+          const fromJson = getEnumMethod(ctx, field.typeName, 'FromJSON');
```

Comment:
Ah yeah, good call passing in the whole `ctx` instead of both `typeMap` + `options`.